Scanner: Generate error on inbalanced RLO/LRO/PDF override markers. (#10326)
ethereum/solidity: Changelog.md

Compiler Features:
 * SMTChecker: Support named arguments in function calls.

Language Features:

>  * Scanner: Generates a parser error when comments do contain an unbalanced or underflowing set of unicode direction override markers (LRO, RLO, PDF)

Reviewer: ```suggestion
 * Scanner: Generate a parser error when comments do contain an unbalanced or underflowing set of unicode direction override markers (LRO, RLO, PDF)
```

 * Code generator: Support copying of nested arrays from calldata to memory.
 * The fallback function can now also have a single ``calldata`` argument (equaling ``msg.data``) and return ``bytes memory`` (which will not be ABI-encoded but returned as-is).
 * Wasm backend: Add ``i32.select`` and ``i64.select`` instructions.

>  * Scanner: Generate a parser error when comments or unicode strings do contain an unbalanced or underflowing set of unicode direction override markers (LRO, RLO, PDF)

Reviewer 2: ```suggestion
 * Scanner: Generate a parser error when comments or unicode strings contain an unbalanced or underflowing set of unicode direction override markers (LRO, RLO, PDF)
```

 * Code generator: Support conversion from calldata slices to memory and storage arrays.
 * The fallback function can now also have a single ``calldata`` argument (equaling ``msg.data``) and return ``bytes memory`` (which will not be ABI-encoded but returned as-is).
 * Wasm backend: Add ``i32.select`` and ``i64.select`` instructions.

>  * Scanner: Generate a parser error when comments or unicode strings do contain an unbalanced or underflowing set of unicode direction override markers (LRO, RLO, PDF).

Reviewer: ```suggestion
 * Scanner: Generate a parser error when comments or unicode strings contain an unbalanced or underflowing set of unicode direction override markers (LRO, RLO, LRE, RLE, PDF).
```

 * Standard-Json: Move the recently introduced ``modelCheckerSettings`` key to ``settings.modelChecker``.
 * Standard-Json: Properly filter the requested output artifacts.

Language Features:

>  * Scanner: Generate a parser error when comments do contain an unbalanced or underflowing set of unicode direction override markers (LRO, RLO, PDF)

Reviewer 2: We already have 'Language Features' above :)

Reviewer 2: Also strings.

Reviewer 3: Fixed it now
